app/controllers/rentals_controller.rb

```diff
@@ -0,0 +1,89 @@
+class RentalsController < ApplicationController
+
+  def checkout
+    video = Video.find_by(id: params[:video_id])
+    customer = Customer.find_by(id: params[:customer_id])
+
+    if video != nil && customer != nil
+      rental = Rental.new(
+        customer_id: customer.id,
+        video_id: video.id
+      )
+      rental.checkout_update(Date.today)
+      if rental.save
+        customer.update_checkout(1)
+        video.update_inventory(-1)
+        rental_data = {
+          customer_id: rental.customer_id,
+          video_id: rental.video_id,
+          due_date: rental.due_date,
+          available_inventory: video.available_inventory,
+          videos_checked_out_count: customer.videos_checked_out_count
+        }
+
+        render json: rental_data.as_json, status: :ok
+        return
+      else
+        render json: {
+          errors: ["Not Found"]},
+          status: :not_found
```

Comment:
There's actually no chance that the request could be bad at this point because the customer_id and video_id have already been verified on line 7. Hence you don't have a unit test that verifies this case.